openwisp_monitoring/monitoring/base/models.py

```diff
@@ -586,7 +586,14 @@ def read(
                 if key == 'time':
                     continue
                 traces.setdefault(key, [])
-                if decimal_places and isinstance(value, (int, float)):
+                if (
+                    # The adaptive feature does unit conversion and approximation
+                    # in the frontend, hence rounding off is skipped here.
+                    # TODO: Update this code to check for 'adaptive_prefix'
+                    not self.config_dict.get('unit', '').startswith('adaptive')
```

Comment:
why this TODO here?

Reply:
In `gsoc22-demo` PR, Aryaman has changed `adaptive_bytes` to `adaptive_prefix`. The current code with work with both `adaptive_bytes` and `adaptive_prefix`. But, later on we might choose to use `adaptive_prefix` only.

Reply:
ok clear now, thanks! We shall create an issue for this to remind ourselves.